lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,25 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      file_full_path = File.realpath("#{DEVICES_JSON_DIR}/#{device}.rb")
+      require file_full_path
```

Comment:
Let's call `File.exist?` here and return `nil` if false instead of calling `File.exist?` in `read_device` so that we can use `file_full_path`.